src/Microsoft.Framework.PackageManager/Utils/NuGetPackageUtils.cs

```diff
@@ -34,20 +37,30 @@ internal static class NuGetPackageUtils
                 // waiting on this lock don't need to install it again.
                 if (createdNewLock && !File.Exists(targetNupkg))
                 {
-                    // Extracting to the {id}/{version} has an issue with concurrent installs - when a package has been partially
-                    // extracted, the Restore Operation can inadvertly conclude the package is available locally and proceed to read
-                    // partially written package contents. To avoid this we'll extract the package to a sibling directory and Move it 
-                    // to the target path.
-                    var extractPath = Path.Combine(Path.GetDirectoryName(targetPath), Path.GetRandomFileName());
-                    Directory.CreateDirectory(extractPath);
-                    targetNupkg = Path.Combine(extractPath, Path.GetFileName(targetNupkg));
-                    using (var nupkgStream = new FileStream(targetNupkg, FileMode.Create, FileAccess.ReadWrite, FileShare.ReadWrite | FileShare.Delete))
+                    var extractPath = targetPath;
+                    if (performingParallelInstalls)
+                    {
+                        // Extracting to the {id}/{version} has an issue with concurrent installs - when a package has been partially
+                        // extracted, the Restore Operation can inadvertly conclude the package is available locally and proceed to read
+                        // partially written package contents. To avoid this we'll extract the package to a sibling directory and Move it 
+                        // to the target path.
+                        extractPath = Path.Combine(Path.GetDirectoryName(targetPath), Path.GetRandomFileName());
+                        targetNupkg = Path.Combine(extractPath, Path.GetFileName(targetNupkg));
+                    }
+
+                    var extractDirectory = Directory.CreateDirectory(targetPath);
```

Comment:
suspect this and a couple of other `targetPath` references should be using `extractPath`.  right now the .nupkg is streamed into a file in the `extractPath` directory.  but files are extracted directly into the `targetPath` directory and the .nupkg is left around in `extractPath`.